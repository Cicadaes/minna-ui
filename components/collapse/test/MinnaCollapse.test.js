'use strict';

const MinnaCollapse = require('../src/MinnaCollapse.html');

describe('MinnaCollapse component', () => {
  it('renders correctly with no props', () => {
    expect.assertions(3);
    function wrapper() {
      const target = document.createElement('div');
      new MinnaCollapse({ target });
      expect(target.querySelector('.collapse-hide')).not.toBeNull();
      expect(target.innerHTML).toMatchSnapshot();
    }
    expect(wrapper).not.toThrow();
  });

  it('collapsed content shows and hides on click', () => {
    expect.assertions(4);
    const target = document.createElement('div');
    const component = new MinnaCollapse({ target });
    expect(component.get().__isOpen).toBeFalsy();
    const button = target.querySelector('button');
    button.click();
    expect(component.get().__isOpen).toBeTruthy();
    button.click();
    expect(component.get().__isOpen).toBeFalsy();
    expect(document.querySelector('collapse-hide')).toBeNull();
  });
});
