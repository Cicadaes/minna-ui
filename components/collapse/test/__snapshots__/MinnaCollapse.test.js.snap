// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinnaCollapse component renders correctly with no props 1`] = `
"<button class=\\"button\\">Expand â–¾</button>

<div class=\\"collapse-hide\\"></div>"
`;
